File: niro/src/main/java/nl/uva/se/sc/niro/gui/QLBaseController.java
package nl.uva.se.sc.niro.gui;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.Button;
import javafx.stage.Stage;

public class QLBaseController {
    @FXML
    public void quitApplication(ActionEvent event) {
        getActiveStage(event).close();
    }

    protected Stage getActiveStage(ActionEvent event) {
        Button closeButton = (Button) event.getSource();
        return (Stage) closeButton.getScene().getWindow();
    }
}

File: niro/src/main/scala/nl/uva/se/sc/niro/gui/controllers/QLBaseController.scala
package nl.uva.se.sc.niro.gui.controllers

import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.stage.Stage

class QLBaseController {
  @FXML
  def quitApplication(event: ActionEvent): Unit = getActiveStage(event).close()

  protected def getActiveStage(event: ActionEvent): Stage = {
    val closeButton = event.getSource.asInstanceOf[Button]
    closeButton.getScene.getWindow.asInstanceOf[Stage]
  }
}